parity-util-mem/src/impls.rs

```diff
@@ -73,3 +69,59 @@ impl<T: MallocSizeOf> MallocSizeOf for RwLock<T> {
 		self.read().size_of(ops)
 	}
 }
+
+#[cfg(test)]
+mod tests {
+	use crate::{allocators::new_malloc_size_ops, MallocSizeOf, MallocSizeOfOps};
+	use smallvec::SmallVec;
+	impl_smallvec!(3);
+
+	#[test]
+	fn test_smallvec_stack_allocated_type() {
+		let mut v: SmallVec<[u8; 3]> = SmallVec::new();
+		let mut ops = new_malloc_size_ops();
+		assert_eq!(v.size_of(&mut ops), 0);
+		v.push(1);
+		v.push(2);
+		v.push(3);
+		assert_eq!(v.size_of(&mut ops), 0);
+		assert!(!v.spilled());
+		v.push(4);
+		assert!(v.spilled(), "SmallVec spills when going beyond the capacity of the inner backing array");
+		assert_eq!(v.size_of(&mut ops), 4); // 4 u8s on the heap
+	}
+
+	#[test]
+	fn test_smallvec_boxed_stack_allocated_type() {
+		let mut v: SmallVec<[Box<u8>; 3]> = SmallVec::new();
+		let mut ops = new_malloc_size_ops();
+		assert_eq!(v.size_of(&mut ops), 0);
+		v.push(Box::new(1u8));
+		v.push(Box::new(2u8));
+		v.push(Box::new(3u8));
+		assert_eq!(v.size_of(&mut ops), 3); // 3 u8s on the heap, boxes are on the stack
```

Comment:
@cheme @ordian This passes on macos, fails on linux where `v.size_of(&mut ops)` returns 72, also fails on linux when using the `jemalloc-global` feature (`size_of` returns 24).
Advice on what to do here?